src/Common/GetPriorityForLoadBalancing.cpp

```diff
@@ -33,19 +33,17 @@ std::function<Priority(size_t index)> GetPriorityForLoadBalancing::getPriorityFu
             get_priority = [offset](size_t i) { return i != offset ? Priority{1} : Priority{0}; };
             break;
         case LoadBalancing::ROUND_ROBIN:
-            if (last_used >= pool_size)
-                last_used = 0;
-            ++last_used;
-            /* Consider pool_size equals to 5
-             * last_used = 1 -> get_priority: 0 1 2 3 4
-             * last_used = 2 -> get_priority: 4 0 1 2 3
-             * last_used = 3 -> get_priority: 4 3 0 1 2
-             * ...
-             * */
-            get_priority = [this, pool_size](size_t i)
+            auto local_last_used = last_used.fetch_add(1) % pool_size;
+
+            get_priority = [pool_size, local_last_used](size_t i)
```

Comment:
Please consider keeping the comment or replacing it with a more comprehensive one, otherwise it's impossible to understand without taking quite a bit of time.

Example from my notes:

| Last used | Pri0 | Pri1 | Pri2 | Pri3 | Pri4 |
|-----------|------|------|------|------|------|
| 0         | 4    | 0    | 1    | 2    | 3    |
| 1         | 3    | 4    | 0    | 1    | 2    |
| 2         | 2    | 3    | 4    | 0    | 1    |
| 3         | 1    | 2    | 3    | 4    | 5    |
| 4         | 0    | 1    | 2    | 3    | 4    |

Reply:
Thanks, yes, will add the comment. Your math is correct 💯